tests/checks/README.rst

```diff
@@ -0,0 +1,14 @@
+Core Go rules tests
```

Comment:
Move to `tests/core/checks`

Reply:
Done, along with everything else in tests/checks/.

Reply:
Done.